packages/wonder-blocks-floating/src/components/floating.tsx

```diff
@@ -1,5 +1,208 @@
+    onOpenChange,
+    strategy = "absolute",
+    testId,
+    // middleware specific
+    hide: hideProp = true,
+    offset: offsetProp = 20,
+    flip: flipProp = true,
+    shift: shiftProp = true,
+    showArrow = true,
+}: FloatingProps) {
+    const arrowRef = React.useRef(null);
+    const prevOpenRef = React.useRef(open ?? false);
+
+    // Calculate the floating styles and context
+    const {refs, floatingStyles, context, middlewareData} = useFloating({
+        open,
+        onOpenChange,
+        placement,
+        strategy,
+        // Ensure the floating element stays in sync with the reference element
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            // Add offset from the reference element
+            offset({mainAxis: offsetProp}),
+            // Flip to the opposite side if there's not enough space
+            flipProp ? flip() : undefined,
+            // Shift along the axis to keep it in view
+            shiftProp
+                ? shift({
+                      padding: boundaryPadding,
+                      crossAxis: true,
+                  })
+                : undefined,
+            showArrow ? arrow({element: arrowRef}) : undefined,
+            hideProp ? hide() : undefined,
+        ],
+    });
+
+    // call onOpenChange when the floating element is opened or closed
+    React.useEffect(() => {
+        // only trigger when the open value changes and is controlled mode
+        if (prevOpenRef.current !== open) {
+            onOpenChange?.(open);
+            // update the previous open value
+            prevOpenRef.current = open;
+        }
+    }, [onOpenChange, open]);
+
+    // Clone the child element and add the ref
+    const trigger = React.useMemo(() => {
+        return React.cloneElement(children, {
+            ref: refs.setReference,
+        });
+    }, [children, refs.setReference]);
+
+    return (
+        <>
+            {trigger}
+            {open && (
+                <StyledDiv
+                    data-testid={testId}
+                    data-placement={placement}
+                    ref={refs.setFloating}
+                    style={[
+                        styles.floating,
+                        floatingStyles,
+                        {
+                            visibility: middlewareData.hide?.referenceHidden
+                                ? "hidden"
+                                : "visible",
+                        },
+                    ]}
+                >
+                    {content}
+                    {showArrow && <Arrow ref={arrowRef} context={context} />}
+                </StyledDiv>
+            )}
+        </>
+    );
 }
+
+const styles = StyleSheet.create({
+    floating: {
+        background: semanticColor.core.background.base.default,
+        border: `solid ${border.width.thin} ${semanticColor.core.border.neutral.subtle}`,
+        borderRadius: border.radius.radius_040,
+        maxInlineSize: 472,
```

Comment:
How did you pick the 472 value!

Reply:
I got it from TooltipBubble, but I don't have a strong opinion about it. Do you think this could have a different value?

Reply:
I think it's fine, I was curious if there was a particular reason for it that'd be helpful to include in a comment!